src/ComposerRequireChecker/DependencyGuesser/GuessFromInstalledComposerPackages.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ComposerRequireChecker\DependencyGuesser;
+
+use Roave\BetterReflection\BetterReflection;
+use Roave\BetterReflection\Identifier\Identifier;
+use Roave\BetterReflection\Identifier\IdentifierType;
+use Roave\BetterReflection\Reflection\ReflectionClass;
+use Roave\BetterReflection\Reflector\DefaultReflector;
+use Roave\BetterReflection\SourceLocator\Type\Composer\Factory\MakeLocatorForInstalledJson;
+use Generator;
+use Roave\BetterReflection\SourceLocator\Type\MemoizingSourceLocator;
+use Roave\BetterReflection\SourceLocator\Type\SourceLocator;
+use function preg_match;
+use function preg_quote;
+use function sprintf;
+use function str_replace;
+use const DIRECTORY_SEPARATOR;
+
+final class GuessFromInstalledComposerPackages implements Guesser
+{
+    private SourceLocator $sourceLocator;
+    private string $pathRegex;
+
+    public function __construct(string $installationPath)
+    {
+        $this->sourceLocator = new MemoizingSourceLocator((new MakeLocatorForInstalledJson())(
+            $installationPath,
+            (new BetterReflection())->astLocator()
+        ));
+
+        $cleanPath = preg_quote(sprintf('%s/vendor', str_replace(DIRECTORY_SEPARATOR, '/', $installationPath)), '@');
+        $this->pathRegex = sprintf('@^%s/(?:composer/\.\./)?([^/]+/[^/]+)/@', $cleanPath);
```

Comment:
I'm not sure for this regex tbh. Is there a way to get the name from the `composer.json` maybe?

Reply:
Getting the name from `composer.json` would be best, since the path will break anyway, due to people doing ugly things like changing the location of the vendor dir :thinking:

Reply:
Is there already a way to get the name from the `composer.json`?

Reply:
each `composer.json` must have a `"name"` property, in order to be valid.

Reply:
So i reused now the regex but to get the composer.json and load it via the JsonLoader already present in this lib.

Im not sure if this is now really better since we are still using a regex but also have now more I/O which impacts performance. But since this is a CI tool i think the performance aspect could be overlooked.